sympy/logic/tests/test_boolalg.py

```diff
@@ -114,6 +114,9 @@ def test_Not():
     assert Not(0) is true
     assert Not(1) is false
     assert Not(2) is false
+    assert Not(A).canonical == Not(A)
+    e = A > 1
+    assert Not(e).canonical == Not(e)
```

Comment:
`Not(e)` is automatically converted to `A <=1` and so this does not test the added code; try `Not(A).canonical`.

Reply:
@smichr I believe I am testing `Not(A).canonical` in the test above (unless I'm misunderstanding your comment?) I really only added the `Not(e).canonical` for completeness, as this is the case that originally triggered the bug when  conversion to `A <= 1` was  suppressed by `with evaluate(False)`. I can remove the case `Not(e)` case, if you want?

Reply:
In the test, however, you don't use `evaluate=False` so you aren't dealing with an expression that is a Not. If you use `Not(A)`, you will be.

Reply:
But it might make more sense to do `return self(getattr(self.args[0], 'canonical', self.args[0]))`

Reply:
Fair enough. Removed.